recipes/autoconf/all/conanfile.py

```diff
@@ -1,122 +1,147 @@
-from conans import AutoToolsBuildEnvironment, ConanFile, tools
-import contextlib
-import os
+from conan import ConanFile
+from conan.tools.env import VirtualBuildEnv
+from conan.tools.files import copy, get, rmdir, apply_conandata_patches, replace_in_file
+from conan.tools.gnu import Autotools, AutotoolsToolchain, AutotoolsDeps
+from conan.tools.layout import basic_layout
+from conan.tools.microsoft import unix_path, is_msvc
 
-required_conan_version = ">=1.33.0"
+required_conan_version = ">=1.51.3"
 
 
 class AutoconfConan(ConanFile):
     name = "autoconf"
+    description = "Autoconf is an extensible package of M4 macros that produce shell scripts to automatically configure software source code packages"
+    license = ("GPL-2.0-or-later", "GPL-3.0-or-later")
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://www.gnu.org/software/autoconf/"
-    description = "Autoconf is an extensible package of M4 macros that produce shell scripts to automatically configure software source code packages"
     topics = ("autoconf", "configure", "build")
-    license = ("GPL-2.0-or-later", "GPL-3.0-or-later")
     settings = "os", "arch", "compiler", "build_type"
 
-    exports_sources = "patches/*"
-
-    _autotools = None
-
-    @property
-    def _source_subfolder(self):
-        return os.path.join(self.source_folder, "source_subfolder")
-
     @property
     def _settings_build(self):
+        # TODO: Remove for Conan v2
         return getattr(self, "settings_build", self.settings)
 
+    @property
+    def win_bash(self):
```

Comment:
Do you want to override [win_bash](https://docs.conan.io/en/latest/reference/conanfile/attributes.html#win-bash)? If yes, I don't think is a good idea. You could set win_bash on `configure(self)` I think. We don't allow public methods, besides those officials, otherwise, you need to set as protected `_win_bash`

Reply:
I got the pattern from the Conan 2 documentation
https://docs.conan.io/en/2.0/reference/conanfile/attributes.html#win-bash

Reply:
Setting it in the configure method is also fine I think

Reply:
let's use configure method then, which is safer:

```python
def configure(self):
    self.win_bash = self._settings_build.os == "Windows"
```

There are two important points:
- We don't allow public methods in CCI, in case a new feature creates a name collision. Many methods from conan.tools were created based on idioms that we had in CCI.
- This new win_bash is still experimental, so in case a breaking change, overriding it could result in side-effects.

Thank you for pointing the documentation, I'll open a PR suggesting a safer approach.

Reply:
Porting your suggestions across all my PRs now.
I also messaged you on Slack with a question regarding a work around for a bug on windows and bash based recipes.

Reply:
I would advice to set win_bash in build_requirements, so that all the subsystem activation logic lies in the same method.

```python
    def build_requirements(self):
        if self._settings_build.os == "Windows":
            if not self.conf.get("tools.microsoft.bash:path", default=False, check_type=bool):
                self.tool_requires("msys2/cci.latest")
            self.win_bash = True
```